### ISSUE TITLE
syntax warning for invalid escape sequence

### ISSUE THREAD

**Issue by FabriceSalvaire** [astral-sh/ty#2900]
### Summary

Ty doesn't warn for
```py
print('\q')
```
Python 3.14:
```
SyntaxWarning: "\q" is an invalid escape sequence. Such sequences will not work in the future. Did you mean "\\q"? A raw string is also an option.
```

### Version

0.18.0

**Comment by AlexWaygood** [comment]
I think this is a diagnostic we would probably want to emit from our parser rather than a ty crate, so it's already covered by https://github.com/astral-sh/ruff/issues/20196 (@ntBre, correct me if I'm wrong there!)

**Comment by carljm** [comment]
It looks like ruff already emits W605 for invalid escape sequences -- but it doesn't look like it's currently implemented in a way that would be shared with ty. I guess that's covered by https://github.com/astral-sh/ruff/issues/20196 ?

**Comment by ntBre** [comment]
Agreed, I think that's covered by https://github.com/astral-sh/ruff/issues/20196. I expected we'd handle this like syntax errors with shared code, but added `needs-design` assuming we'd at least want to discuss it.
